common/components/AddStreetModal/index.tsx

```diff
@@ -30,20 +42,32 @@ const AddStreetModal: FC<Props> = ({ closeModal, edit }) => {
     }
   }
 
+  const getTitle = () => {
+    if (edit) return 'Редагування адреси'
+    if (preview) return 'Перегляд адреси'
+    return 'Додавання адреси'
+  }
+
+  useEffect(() => {
+    form.setFieldsValue(currentStreet)
+  }, [form, currentStreet])
+  console.log(edit)
   return (
     <Modal
-      title={!edit && 'Додавання адреси'}
+      title={getTitle()}
       onOk={handleSubmit}
       changesForm={() => form.isFieldsTouched()}
       onCancel={() => {
         form.resetFields()
         closeModal()
       }}
-      okText={!edit && 'Додати'}
-      cancelText={edit ? 'Закрити' : 'Відміна'}
+      okText={preview ? 'Закрити' : 'Додати'}
+      cancelText={preview ? '' : edit ? 'Закрити' : 'Відміна'}
```

Comment:
Will be better to swap buttons logic

`edit` - show both buttons "Закрити", "Зберегти"
`create` - show both buttons "Закрити", "Зберегти"
`preview` - show ONLY cancel button "Закрити"

```cancelButtonProps={preview ? { disabled: true } : {}}``` can also be removed if the logic above will be realized
```cancelButtonProps={preview ? { disabled: true } : {}}``` also duplicated in Modal component directly as ```cancelButtonProps={preview ? { style: { display: 'none' } } : cancelButtonProps}```, so it can be removed at both (here and from Modal inner component